symbulate/tests/test_distributions.py

```diff
@@ -17,18 +17,460 @@ def test_p_one(self):
     def test_sum(self):
         X = RV(Bernoulli(.4) ** 5)
         sims = X.apply(sum).sim(Nsim)
+        key_list = [keys for keys in sims.tabulate()]
         obs = list(sims.tabulate().values())
-        exp = Nsim * stats.binom(n=5, p=.4).pmf(range(6))
+        exp = [Nsim * stats.binom(n = 5, p = .4).pmf(k) for k in key_list]
+        pval = stats.chisquare(obs, exp).pvalue
+        self.assertTrue(pval > 0.01)
+
+    def test_Bernoulli_Binomial_n_1(self):
+        exp, obs, new_key = [], [], []    
+        X = RV(Bernoulli(.4))
+        sims = X.sim(Nsim)
+        key_list = list(sims.tabulate().keys())
+        for k in key_list:
```

Comment:
I don't like this behavior where you only iterate over the observed values. You should know what the support of the distribution is. (In this case, it only takes on the values 0, 1.) You should iterate over those values, rather than relying on the simulation to tell you what the values should be.

Please fix this everywhere below. (For some distributions, like Poisson, the support is infinite because it can take on values 0, 1, 2, .... You'll need to choose a reasonable upper bound for these distributions.)